src/elements/ProjectSettingsGeneral/ProjectSettingsGeneral.js

```diff
@@ -0,0 +1,266 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import ProjectSettingsGeneralView from './ProjectSettingsGeneralView'
+
+import {
+  ARTIFACT_PATH,
+  SOURCE_URL
+} from '../../components/ProjectSettings/projectSettings.util'
+import { KEY_CODES } from '../../constants'
+import projectsApi from '../../api/projects-api'
+import projectsAction from '../../actions/projects'
+
+import './projectSettingsGeneral.scss'
+
+const ProjectSettingsGeneral = ({
+  match,
+  projectStore,
+  setProjectParams,
+  setProjectSettings
+}) => {
+  const [editProject, setEditProject] = useState({
+    source: {
+      value: null,
+      isEdit: false
+    },
+    artifact_path: {
+      value: null,
+      isEdit: false
+    }
+  })
+
+  const inputRef = React.createRef()
+
+  const generalParams = useMemo(
+    () =>
+      projectStore.project.data?.spec.params
+        ? Object.entries(projectStore.project.data?.spec.params).map(elem => ({
+            key: elem[0],
+            value: elem[1]
```

Comment:
Could use destructuring here to avoid using `0` and `1` indexes which might be less readable than giving the meaningful names of `key` and `value`:
```suggestion
        ? Object.entries(projectStore.project.data?.spec.params).map(([key, value]) => ({
            key,
            value
```